### PR TITLE
feat: add configuration option to read_gbq

### LINKED ISSUE #384: Support for query parameterization
I was wondering if there were plans to expose the ability to pass bound parameters (either named or qmark style) when constructing constructing a bigframes DataFrame e.g. via `Session.read_gbq_query`? My understanding is they would need to be thread through to the [`queryParameters` field](https://cloud.google.com/bigquery/docs/reference/rest/v2/Job#JobConfigurationQuery.FIELDS.query_parameters) on the `QueryJobConfig` that gets constructed.

I realize it's probably a bit of a niche use-case since this is a code-based API but for context we were looking to integrate support for bigframes in a product that allows users to build Python notebook-based interactive apps which may be used by external/untrusted users (not the ones writing the app/code) and so our SQL functionality relies on using parameterization whenever these kinds of inputs are involved.

### PR DESCRIPTION
**Pull request by Genesis929**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes https://github.com/googleapis/python-bigquery-dataframes/issues/384 🦕

### CODE REVIEW

File: tests/system/small/test_session.py [hunk 1]
```diff
@@ -20,6 +20,8 @@
 import typing
 from typing import List
 
+import google
+from google.api_core.exceptions import InternalServerError
```
Comment (shobsi) [inline tests/system/small/test_session.py:24]: import module not class go/pystyle#imports

File: tests/system/small/test_session.py [hunk 2]
```diff
@@ -353,6 +355,45 @@ def test_read_gbq_table_wildcard_with_filter(session: bigframes.Session):
...
+            "query": {
+                "useQueryCache": True,
+                "maximumBytesBilled": "1000000000",
+                "timeoutMs": 10000,
+            }
+        },
+        pytest.param(
+            {"query": {"useQueryCache": True, "timeoutMs": 50}},
+            marks=pytest.mark.xfail(
+                raises=google.api_core.exceptions.BadRequest,
```
Comment (shobsi) [inline tests/system/small/test_session.py:371]: curious why timeout 50ms is a bad request?

File: tests/system/small/test_session.py [hunk 3]
```diff
@@ -353,6 +355,45 @@ def test_read_gbq_table_wildcard_with_filter(session: bigframes.Session):
...
+            ),
+        ),
+        pytest.param(
+            {"query": {"useQueryCache": False, "maximumBytesBilled": "100"}},
+            marks=pytest.mark.xfail(
+                raises=InternalServerError,
```
Comment (shobsi) [inline tests/system/small/test_session.py:377]: this too, why internal server error? would be nice to add a comment for the reason

File: tests/system/small/test_session.py (hunk 2 above)
Comment (Genesis929) [reply tests/system/small/test_session.py:371]: It's an error that the time set is too short, this is a quick way to prove that the parameter works.

File: bigframes/session/__init__.py [hunk 4]
```diff
@@ -376,7 +386,7 @@ def _query_to_destination(
         query: str,
         index_cols: List[str],
         api_name: str,
-        use_cache: bool = True,
+        configuration: dict = {"query": {"useQueryCache": True}},
```
[the remark this answers is not in the document]
Comment (Genesis929) [reply bigframes/session/__init__.py:389]: Done

File: tests/system/small/test_session.py (hunk 1 above)
Comment (Genesis929) [reply tests/system/small/test_session.py]: Done

File: tests/system/small/test_session.py (hunk 3 above)
Comment (Genesis929) [reply tests/system/small/test_session.py]: Updated reason.

Comment (shobsi) [reply tests/system/small/test_session.py]: Thanks!

File: tests/system/small/test_session.py (hunk 2 above)
Comment (shobsi) [reply tests/system/small/test_session.py:380]: Makes sense, thanks!

**Review by shobsi — COMMENTED** [review]
Please wait for all checks to pass before merging

**Review by shobsi — APPROVED** [review]

**Pull request merged**